### PR TITLE
Fix target directory for cloned ros1_bridge repository to ensure it is built in final image

### PR DESCRIPTION
**Pull request by JonSmet**
# PR Details
## Description

This PR updates `checkout.bash` to change the target directory in which the `ros1_bridge` repo is cloned into when the `BRANCH` environment variable is set to `develop`. Prior to this change, when `BRANCH` was set to `develop`, the `ros1_bridge` repo was cloned into /home/carma/src, but the `install.sh` script only builds ROS 2 packages located at ~/.base-image/ros2_msgs_ws

## How Has This Been Tested?

CI (docker/docker action)

## Types of changes

- [x] Defect fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that cause existing functionality to change)

## Checklist:

- [ ] I have added any new packages to the sonar-scanner.properties file
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING**](https://github.com/usdot-fhwa-stol/carma-platform/blob/develop/Contributing.md) document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: docker/checkout.bash [hunk 1]
```diff
@@ -45,16 +45,16 @@ cd ${dir}/src
 if [[ "$BRANCH" == "develop" ]]; then
       cd /home/carma/
       git clone https://github.com/usdot-fhwa-stol/autoware.ai.git --depth 1 --branch carma-"$BRANCH"
-      cd ${dir}/src 
+      cd /home/carma/.base-image/workspace/src
```
Comment (paulbourelly999) [inline docker/checkout.bash:48]: Is this the only change. If so, I think the calling script just needs to specify this directory. We probably also have to fix the else statement to clone into this directory too. When I say "this directory" I mean the one provided as an argument to the script `-r <dir>` and set to the local variable ${dir}

Comment (JonSmet) [reply docker/checkout.bash]: Addressed in [472a285](https://github.com/usdot-fhwa-stol/carma-msgs/pull/234/commits/472a28514a86bc099e1c8a88374a0c4f8c1fc486)

**Comment by MishkaMN** [comment]
This build is finishing fine, it's just that the github action is running out of space. Tested on this PR
https://github.com/usdot-fhwa-stol/carma-msgs/pull/235

**Comment by JonSmet** [comment]
Closing this PR since the change is being included in #237

**Pull request closed without merging**